distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/APMJvmOptions.java

```diff
@@ -216,26 +216,38 @@ private static Map<String, String> extractDynamicSettings(Map<String, String> pr
     static Map<String, String> extractApmSettings(Settings settings) throws UserException {
         final Map<String, String> propertiesMap = new HashMap<>();
 
-        final Settings agentSettings = settings.getByPrefix("tracing.apm.agent.");
-        agentSettings.keySet().forEach(key -> propertiesMap.put(key, String.valueOf(agentSettings.get(key))));
+        // tracing.apm.agent. is deprecated by telemetry.agent.
+        final String telemetryAgentPrefix = "telemetry.agent.";
+        final String tracingAgentPrefix = "tracing.apm.agent.";
```

Comment:
i wonder if it would be readable by naming those two `telemetryAgentPrefix` and `deprecatedTelemetryAgentPrefix`

Reply:
Changed.